### ISSUE TITLE
Can't merge tickets from ticket view

### ISSUE THREAD

**Issue by berkenpies** [osTicket/osTicket#5313]
We're having difficulties merging tickets.

Problems occurs when you go directly to a ticket and use the merge ticket function from the options menu.

**Steps to reproduce:**

- Click on any ticket in the "Open" ticket queue. Haven't tried it with other queues but I assume problem is most likely not related to queues.
 
- Go to "Merge tickets" (Option/Configuration icon -> "Merge tickets").
 ![ticket_merge](https://user-images.githubusercontent.com/21110769/71714483-61981c80-2e0e-11ea-8a72-e95dca2fcebf.png)
 
- Use "Select ticket" to find an other ticket you wish to merge with and click "Add a ticket". Now you have 2 tickets in your overview.
![ticket_merge_2tickets](https://user-images.githubusercontent.com/21110769/71714730-4a0d6380-2e0f-11ea-9098-1e9f7407742e.png)

- Click "Save changes".

Nothing will happen after that. Saving results in a (silent) 500 error and osTicket will redirect you to the ticket overview.

The actual error from the PHP-FPM error log:

`2020/01/03 09:11:05 [error] 29388#0: *231869 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error: Call to a member function getMergeType() on null in <osticket_path>/include/class.ticket.php:2531
Stack trace:
#0 <osticket_path>/include/ajax.tickets.php(389): Ticket::merge(Array)
#1 <osticket_path>/include/class.dispatcher.php(145): TicketsAjaxAPI->updateMerge('<id>')
#2 <osticket_path>/include/class.dispatcher.php(38): UrlMatcher->dispatch('<id>/merge', Array)
#3 <osticket_path>/include/class.dispatcher.php(120): Dispatcher->resolve('<id>/merge', Array)
#4 <osticket_path>/include/class.dispatcher.php(38): UrlMatcher->dispatch('/tickets/<id>/...', NULL)
#5 <osticket_path>/scp/ajax.php(308): Dispatcher->resolve('/tickets/<id>/...')
#6 {main}`

Using the merge function from the (open) ticket overview seems to be working fine (yay!). With this I mean selecting tickets by using the checkboxes and the brand new merge icon:
![ticket_merge_from_overview](https://user-images.githubusercontent.com/21110769/71715120-9c9b4f80-2e10-11ea-9ce6-e8c174d424c8.png)

**Server components (if needed):**
nginx/1.17.7
PHP 7.2.24
mysql  Ver 15.1 Distrib 10.3.11-MariaDB

Just let me know if more details are needed. :)

**Comment by JediKev** [comment]
@berkenpies 

I cannot replicate the same issue. Are you all the way up to date with v1.14.1? If not, please upgrade and retest.

Cheers.

**Comment by jhayn49** [comment]
running into the exact same issue with merges.

[Fri Jan 03 09:49:37.597024 2020] [php7:error] [pid 13348] [client 10.x.x.x:62386] PHP Fatal error:  Uncaught Error: Call to a member function getMergeTack trace:\n#0 <osticket_path>/include/ajax.tickets.php(389): Ticket::merge(Array)\n#1 <osticket_path>/class.dispatcher.pinclude/class.dispatcher.php(38): UrlMatcher->dispatch('15924/merge', Array)\n#3 <osticket_path>/include/class.dispatcher.php(120): Dispatcher-ispatcher.php(38): UrlMatcher->dispatch('/tickets/15924/...', NULL)\n#5 <osticket_path>/scp/ajax.php(308): Dispatcher->resolve('/tickets/15924/.php on line 2531



![image](https://user-images.githubusercontent.com/19733584/71742653-79809800-2e30-11ea-86c0-7c9bae718044.png)

**Comment by JediKev** [comment]
@jhayn49 

I assume that you copied the error wrong or something because `getMergeTack ` is not a valid function in the codebase. Try to find the full error in your logs to ensure that it does not say `getMergeTack `. If it does then search the codebase for `getMergeTack `. If this function exists you must have customizations or something installed as I mentioned before, this is not a valid function in the core codebase.

Cheers.

**Comment by jhayn49** [comment]
I went back to replicate this and now I can't. It's merging correctly for me at the ticket view screen. Thanks for the quick reply. It looks like I'm okay now.

This was the error I had.

[Fri Jan 03 09:44:35.541116 2020] [php7:error] [pid 12667] [client :62310] PHP Fatal error:  Uncaught Error: Call to a member function getMergeType() on null in /var/www/html/dev/ack trace:\n#0 /var/www/html/dev/tickets_dev/include/ajax.tickets.php(389): Ticket::merge(Array)\n#1 /var/www/html/dev/tickets_dev/include/class.dispatcher.php(145): TicketsAjaxAPI->updateMerginclude/class.dispatcher.php(38): UrlMatcher->dispatch('15893/merge', Array)\n#3 /var/www/html/dev/tickets_dev/include/class.dispatcher.php(120): Dispatcher->resolve('15893/merge', Array)\n#4 ispatcher.php(38): UrlMatcher->dispatch('/tickets/15893/...', NULL)\n#5 /var/www/html/dev/tickets_dev/scp/ajax.php(308): Dispatcher->resolve('/tickets/15893/...')\n#6 {main}\n  thrown in /var/.php on line 2531, referer: https://[URL]/dev/scp/tickets.php?id=15893

**Comment by stevland** [comment]
For the record, I am encountering the exact same issue as the OP: merging tickets works from the Open Ticket page, but from the Ticket View page it fails silently and produces this Apache error:

`mod_fcgid: stderr: PHP Fatal error: Call to a member function getMergeType() on null in /var/www/vhosts/redacted/include/class.ticket.php on line 2531, referer: https://redacted/scp/tickets.php?id=91`


osTicket Version | v1.14.1 (f1e9e88) —  Up to date
Web Server Software | Apache
MySQL Version | 5.5.64
PHP Version | 5.6.40

**Comment by stevland** [comment]
I anticipate that @JediKev will point out that my PHP is outdated, which is certainly true.

So I've updated to 7.3.13. 

Merging tickets from the Ticket View page still fails silently, although instead of the one error I now receive a flurry of them:

```
2020-01-04 14:24:44 | Error | xx.xx.xx.xx | POST /o/1.14.1/scp/ajax.php/tickets/230/merge HTTP/1.1 | Apache SSL/TLS access
2020-01-04 14:24:44 | Warning | xx.xx.xx.xx | mod_fcgid: stderr: PHP Fatal error: Uncaught Error: Call to a member function getMergeType() on null in /var/www/vhosts/redacted/include/class.ticket.php:2531, referer: https://redacted/scp/tickets.php?id=230 | Apache error
2020-01-04 14:24:44 | Warning | xx.xx.xx.xx | mod_fcgid: stderr: Stack trace:, referer: https://redacted/scp/tickets.php?id=230 | Apache error
2020-01-04 14:24:44 | Warning | xx.xx.xx.xx | mod_fcgid: stderr: #0 /var/www/vhosts/redacted/include/ajax.tickets.php(389): Ticket::merge(Array), referer: https://redacted/scp/tickets.php?id=230 | Apache error
2020-01-04 14:24:44 | Warning | xx.xx.xx.xx | mod_fcgid: stderr: #1 /var/www/vhosts/redacted/include/class.dispatcher.php(145): TicketsAjaxAPI->updateMerge('230'), referer: https://redacted/scp/tickets.php?id=230 | Apache error
2020-01-04 14:24:44 | Warning | xx.xx.xx.xx | mod_fcgid: stderr: #2 /var/www/vhosts/redacted/include/class.dispatcher.php(38): UrlMatcher->dispatch('230/merge', Array), referer: https://redacted/scp/tickets.php?id=230 | Apache error
2020-01-04 14:24:44 | Warning | xx.xx.xx.xx | mod_fcgid: stderr: #3 /var/www/vhosts/redacted/include/class.dispatcher.php(120): Dispatcher->resolve('230/merge', Array), referer: https://redacted/scp/tickets.php?id=230 | Apache error
2020-01-04 14:24:44 | Warning | xx.xx.xx.xx | mod_fcgid: stderr: #4 /var/www/vhosts/redacted/include/class.dispatcher.php(38): UrlMatcher->dispatch('/tickets/230/me...', NULL), referer: https://redacted/scp/tickets.php?id=230 | Apache error
2020-01-04 14:24:44 | Warning | xx.xx.xx.xx | mod_fcgid: stderr: #5 /var/www/vhosts/redacted/scp/ajax.php(308): Dispatcher->resolve('/tickets/230/me...'), referer: https://redacted/scp/tickets.php?id=230 | Apache error
2020-01-04 14:24:44 | Warning | xx.xx.xx.xx | mod_fcgid: stderr: #6 {main}, referer: https://redacted/scp/tickets.php?id=230 | Apache error
2020-01-04 14:24:44 | Warning | xx.xx.xx.xx | mod_fcgid: stderr: thrown in /var/www/vhosts/redacted/include/class.ticket.php on line 2531, referer: https://redacted/scp/tickets.php?id=230
```

**Comment by JediKev** [comment]
@stevland 

Your assumption of my response is incorrect. This shouldn’t be a PHP version issue. Instead I think it will be code related. I will need to do some testing to try and replicate this issue.

Cheers.

**Comment by berkenpies** [comment]
@JediKev Thanks for your quick response. I double-checked the version(s);

osTicket Version:
v1.14.1 (f1e9e88)

Web Server Software:
nginx/1.17.7

MySQL Version:
10.3.11

PHP Version:
7.2.24

**Comment by JediKev** [comment]
@berkenpies @stevland

Okay, so your issue is you are clicking on the result that contains the Subject line and not the Ticket Number. The Subject line appearing in the typeahead results is actually a bug and shouldn't be used. If you click the Ticket Number instead you should be able to merge correctly.

We are going to create a patch for the subject line in results issue and post a link here once complete.

Cheers.

**Comment by JediKev** [comment]
@berkenpies @stevland 

Here you guys go! 😁
- https://github.com/osTicket/osTicket/pull/5386

Cheers.

**Comment by berkenpies** [comment]
@JediKev seems to do the trick! I manually applied the fixes from #5386 to our setup and that seems to fix things. :smile:
